nemo/core/neural_graph/graph_inputs.py

```diff
@@ -0,0 +1,188 @@
+# -*- coding: utf-8 -*-
+
+# =============================================================================
+# Copyright (c) 2020 NVIDIA. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+from collections.abc import MutableMapping
+
+from nemo.core.neural_types import NeuralType
+from nemo.utils import logging
+from nemo.utils.connection import StepModulePort
+
+
+class GraphInput(object):
```

Comment:
Is a GraphInput to a NeuralGraph what a NmTensor is to a NeuralModule?

Reply:
No. Bot GraphInputs and GraphOutputs serve binding purposes and operate _on top of_ NmTensor.